kinto_http/tests/test_functional_async.py

```diff
@@ -0,0 +1,621 @@
+
+
+@pytest.mark.asyncio
+async def test_collection_creation_if_not_exists(functional_setup):
+    client = functional_setup
+    await client.create_bucket(id="mozilla")
+    await client.create_collection(id="payments", bucket="mozilla")
+    # Should not raise.
+    await client.create_collection(id="payments", bucket="mozilla", if_not_exists=True)
+
+
+@pytest.mark.asyncio
+async def test_collection_list(functional_setup):
+    client = functional_setup
+    await client.create_bucket(id="mozilla")
+    await client.create_collection(id="receipts", bucket="mozilla")
+    await client.create_collection(id="assets", bucket="mozilla")
+
+    # The returned collections should be strings.
+    collections = await client.get_collections(bucket="mozilla")
+    assert len(collections) == 2
+
+    assert set([coll["id"] for coll in collections]) == set(["receipts", "assets"])
+
+
+@pytest.mark.asyncio
+async def test_collection_deletion(functional_setup):
+    client = functional_setup
+    await client.create_bucket(id="mozilla")
+    await client.create_collection(id="payments", bucket="mozilla")
+    await client.delete_collection(id="payments", bucket="mozilla")
+    assert len(await client.get_collections(bucket="mozilla")) == 0
+
+
+@pytest.mark.asyncio
+async def test_collection_deletion_if_exists(functional_setup):
+    client = functional_setup
+    await client.create_bucket(id="mozilla")
+    await client.create_collection(id="payments", bucket="mozilla")
+    await client.delete_collection(id="payments", bucket="mozilla")
+    await client.delete_collection(id="payments", bucket="mozilla", if_exists=True)
+
+
+@pytest.mark.asyncio
+async def test_collection_deletion_can_still_raise_errors(functional_setup, mocker: MockerFixture):
+    client = functional_setup
+    error = KintoException("An error occured")
+    mocker.patch.object(client.session, "request", side_effect=error)
+    with pytest.raises(KintoException):
+        await client.delete_collection(id="payments", bucket="mozilla", if_exists=True)
+
+
+@pytest.mark.asyncio
+async def test_collections_deletion(functional_setup):
+    client = functional_setup
+    await client.create_bucket(id="mozilla")
+    await client.create_collection(id="amo", bucket="mozilla")
+    await client.create_collection(id="blocklist", bucket="mozilla")
+    await client.delete_collections(bucket="mozilla")
+    assert len(await client.get_collections(bucket="mozilla")) == 0
+
+
+@pytest.mark.asyncio
+async def test_collections_deletion_when_no_collections_exist(functional_setup):
+    client = functional_setup
+    await client.create_bucket(id="mozilla")
+    deleted_collections = await client.delete_collections(bucket="mozilla")
+    assert len(deleted_collections) == 0
+
+
+@pytest.mark.asyncio
+async def test_record_creation_and_retrieval(functional_setup):
+    client = Client(
+        server_url=SERVER_URL, auth=DEFAULT_AUTH, bucket="mozilla", collection="payments"
+    )
```

Comment:
I wonder if we couldn't use the fixture here as well... Same for the ones below. The idea would be to move `SERVER_URL` and `DEFAULT_AUTH` to `conftest.py` instead of importing it from here